realm/src/main/java/io/realm/Realm.java

```diff
@@ -895,6 +895,45 @@ public void writeCopyTo(File destination) throws IOException {
         return result;
     }
 
+    /**
+     * Converts a RealmObject that has been created using it's standard constructors instead of
+     * {@link #createObject(Class)} to a RealmObject that is managed by the Realm.
+     *
+     * To persist any further changes, the returned object must be be used.
+     *
+     * @param object RealmObject to copy to the Realm.
+     * @return The converted RealmObject that has all its properties managed by the Realm.
+     *
+     * @throws io.realm.exceptions.RealmException if object has already been added to the Realm.
+     */
+    public <E extends RealmObject> E copyToRealm(E object) {
+
+
+        return null;
+    }
+
+    /**
+     * Converts a list of RealmObjects that has been created using their standard constructors
+     * instead of {@link #createObject(Class)} to proper RealmObjects that are persisted inside
+     * Realm.
+     *
+     * @param objects RealmObjects to copy to the Realm.
+     * @return A list of the he converted RealmObject that all has their properties managed by the Realm.
+     *
+     * @throws io.realm.exceptions.RealmException if any of the objects has already been added to Realm.
+     */
+    public <E extends RealmObject> List<E> copyToRealm(List<E> objects) {
```

Comment:
Since we simply iterate throw the objects and we don't need to know the size of the iterable, then I think `Iterable` is the right choice for the argument. If we are able to provide a zero-cost implementation of `size()` for the returned object then we should use `Collection` as a return type, otherwise `Iterable`.